src/guides/v2.3/config-guide/config/config-create.md

```diff
@@ -21,38 +21,84 @@ Your new `events.xml` is automatically collected from your module and merged wit
+
+{:.bs-callout-info}
+If new modules have a `search.xml` file, they will be merged with your file when it loads.
+
+### Examples of use  
+
+To create a new configuration type:
+
+1. Create your XSD file
+2. Create your XML file
+3. Define your configuration object in your `di.xml` 
+
+The following example from the Magento_Sales module's [di.xml]({{ site.mage2bloburl }}/{{ page.guide_version }}/app/code/Magento/Sales/etc/di.xml) illustrates how a configuration object should look like.
+
+```xml
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
+
+    <type name="Magento\Sales\Model\Order\Pdf\Config\Reader">
+        <arguments>
+            <argument name="fileName" xsi:type="string">pdf.xml</argument>
+            <argument name="converter" xsi:type="object">Magento\Sales\Model\Order\Pdf\Config\Converter</argument>
+            <argument name="schemaLocator" xsi:type="object">Magento\Sales\Model\Order\Pdf\Config\SchemaLocator</argument>
+        </arguments>
+    </type>
+
+    <virtualType name="pdfConfigDataStorage" type="Magento\Framework\Config\Data">
+        <arguments>
+            <argument name="reader" xsi:type="object">Magento\Sales\Model\Order\Pdf\Config\Reader</argument>
+            <argument name="cacheId" xsi:type="string">sales_pdf_config</argument>
+        </arguments>
+    </virtualType>
+
+    <type name="Magento\Sales\Model\Order\Pdf\Config">
+        <arguments>
+            <argument name="dataStorage" xsi:type="object">pdfConfigDataStorage</argument>
+        </arguments>
+    </type>
+</config>
+```
+* The first type node sets the Reader's filename, associated `Converter` and `SchemaLocator` classes.
+* Then, the `pdfConfigDataStorage` virtual type node attaches the reader class to an instance of [Magento\Framework\Config\Data]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/internal/Magento/Framework/Config/Data.php). 
+* And finally, the last type node attaches that config data virtual type to the [Magento\Sales\Model\Order\Pdf\Config]({{ site.mage2bloburl }}/{{ page.guide_version }}/app/code/Magento/Sales/Model/Order/Pdf/Config.php) class, which is used for actually reading values in from those [pdf.xml]({{ site.mage2bloburl }}/{{ page.guide_version }}/app/code/Magento/Sales/etc/pdf.xml) files.
 
-   If other modules have a `search.xml` file, they are merged with your file when it loads.
 
-To create a new configuration type, extend the `\Magento\Framework\Config\ReaderInterface`, which is [Magento\Framework\Config\Reader\Filesystem]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/internal/Magento/Framework/Config/Reader/Filesystem.php) to provide the following parameters:
+4. Define a reader by extending [Magento\Framework\Config\Reader\Filesystem]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/internal/Magento/Framework/Config/Reader/Filesystem.php) class to provide the following parameters:
```

Comment:
Hi @diazwatson. The only doubtful point that is left is why we have removed the description of all parameters accepted by the `ReaderInterface`? below. We are still referencing these params here (above)

> to provide the following parameters:

But then goes an example and no parameter description anymore. 

And thank you for providing the detailed info!

Reply:
Hi @rogyar, thanks for taking the time to review this.

I removed that intentionally because I think the original text is a bit confusing.

For instance, it says developers should **extend** an interface but we all know _an interface should be implemented_ no extended.

Then it states that developers should provide all these parameters but in reality if all we need is to create a custom file and we are going to extend `\Magento\Framework\Config\Reader\Filesystem` (rather than implement the interface) we only need to rewrite the protected attribute `$_idAttributes` to map the nodes in the xml file.

As a proof, we can see the Reader in the core used as an example:
```
Model/Order/Pdf/Config/Reader.php
```

In this reader, only `$_idAttributes` has been modified to provide a mapping for the xml file nodes.

In this PR I am still mentioning the Interface for those who want to create their own version of the reader.

I have change the paragraphs a bit trying to make this more clear.

Please let me know what you think

Reply:
It makes sense. Thank you